### PR TITLE
Auto retry on MDM Clear conflicts

### PR DESCRIPTION
**Pull request by jamesagnew**

### CODE REVIEW

File: hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/clear/MdmClearStep.java [hunk 1]
```diff
@@ -72,6 +74,8 @@ public class MdmClearStep implements IJobStepWorker<MdmClearJobParameters, Resou
 	public RunOutcome run(@Nonnull StepExecutionDetails<MdmClearJobParameters, ResourceIdListWorkChunkJson> theStepExecutionDetails, @Nonnull IJobDataSink<VoidModel> theDataSink) throws JobExecutionFailedException {
 
 		SystemRequestDetails requestDetails = new SystemRequestDetails();
+		requestDetails.setRetry(true);
+		requestDetails.setMaxRetries(100);
```
Comment (tadgh) [inline hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/clear/MdmClearStep.java:78]: Question: was this our actual fix in master? This is tagged as a backport, but I don't actually see it in master yet. Either way, it's fine. i'm just slightly worried about the 100 retries in a non-recoverable scenario (will take a while for the job to fail out). But its fine for now, we can re-evaluate if it becomes a problem.

**Review by tadgh — APPROVED** [review]

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_3_0/4398-retry-on-mdm-clear-conflicts.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+type: fix
+backport: 6.2.3
+title: "The $mdm-clear operation sometimes failed with a constraint error when running in a heavily
+  multithreaded environment. This has been fixed."
```
Comment (tadgh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_3_0/4398-retry-on-mdm-clear-conflicts.yaml:5]: issue: move this to 6_4_0 not 6_3_0

File: hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/clear/MdmClearStep.java (hunk 1 above)
Comment (jamesagnew) [reply hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/clear/MdmClearStep.java:78]: I'm gonna do a mergeback of all of these changes into master after this stuff merges.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_3_0/4398-retry-on-mdm-clear-conflicts.yaml (hunk 2 above)
Comment (jamesagnew) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_3_0/4398-retry-on-mdm-clear-conflicts.yaml:5]: Will do

**Pull request merged**